packages/ember-runtime/lib/computed/array_computed.js

```diff
@@ -12,6 +12,9 @@ var a_slice = [].slice;
 function ArrayComputedProperty() {
   var cp = this;
 
+  var options = arguments[arguments.length - 1];
+  options._isArrayComputed = true;
```

Comment:
You can set on `this` instead of grabbing `options` here.

Something like:

```javascript
this._isArrayComputed = true;

ReduceComputedProperty.apply(this, arguments);
```